src/SwedbankPay.Sdk/Consumers/ConsumersResource.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace SwedbankPay.Sdk.Consumers
+{
+    internal class ConsumersResource : ResourceBase, IConsumersResource
+    {
+        public ConsumersResource(SwedbankPayHttpClient swedbankPayHttpClient)
+            : base(swedbankPayHttpClient)
+        {
+        }
+
+
+        /// <summary>
+        ///     Retrieve Consumer Billing Details.
+        ///     When the payer has been identified through checkin you can retrieve the consumers billing details with the url
+        ///     received through the event OnBillingDetailsAvailable.
+        /// </summary>
+        /// <param name="url"></param>
+        /// <returns></returns>
+        public async Task<BillingDetails> GetBillingDetails(Uri url)
+        {
+            if (url == null)
+                throw new ArgumentNullException(nameof(url), "url cannot be null or whitespace");
```

Comment:
The check says nothing about whitespace so I'd suggest removing that comment. Also the exception type is quite self-descriptive so in my opinion there's no need for a comment at all.